graphql/parser.go

```diff
@@ -36,7 +36,7 @@ func valueToJson(value ast.Value, vars map[string]interface{}) (interface{}, err
 	case *ast.Variable:
 		actual, ok := vars[value.Name.Value]
 		if !ok {
```

Comment:
As we discussed irl, there is a new bug introduced in this PR that we think we are okay with ignoring for now.

That is, if a variable is marked optional in the operation definition, we do not check for whether or not it is passed in, which goes against the spec. We think this is okay because if the variable was actually required (i.e. used at some location), then the executor will get unhappy about it downstream.

Reply:
filed an issue here: https://github.com/samsarahq/thunder/issues/73

Reply:
Awesome. Thanks for writing the issue.